suricata.yaml.in

```diff
@@ -660,6 +660,56 @@ af-packet:
     #use-mmap: no
     #tpacket-v3: yes
 
+dpdk:
+  eal-params:
+    proc-type: primary
+    a: 0000:3b:00.0
+    l: 0-1
+
+  # DPDK capture support
+  # Currently queues should be assigned to cores in ratio in 1:1 ratio
+  interfaces:
+    - interface: 0000:3b:00.0 # PCIe address of the NIC port
+      threads: auto # or number of threads - auto takes all lcores, no thread entry takes all cores
+      rx-queues: auto # or number of queues - auto assigns each core 1 queue
+      tx-queues: auto # or number of queues - auto assigns each core 1 queue
+      promisc: true
+      multicast: true # enables also detection on multicast packets
+      checksum-checks: true # if Suricata should validate checksums
+      checksum-checks-offload: true # if possible offload checksum validation to the NIC
+      mtu: 3000 # Set MTU of the device in bytes # default 1526
+      mempool-size: 65356 # 64*1024
```

Comment:
Since your current code is focusing on worker mode, mempool elements depends upon `number of NIC * total number of rx queue`. Size of each elements depends upon JUMBO frame enablement. Hence rework this logic.

Reply:
ahh, you have not understood the problem. please check details of multi-segment receive and Jumbo frame support in DPDK to better understand the issue. Rework